src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,143 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exceptions.RegistrationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final String VALID_LOGIN = "ValidLogin";
+    private static final String EDGE_LOGIN = "EdgeLo";
+    private static final String INVALID_LOGIN = "invL";
+    private static final String VALID_PASSWORD = "password";
+    private static final String EDGE_PASSWORD = "EdgePs";
+    private static final String INVALID_PASSWORD = "invP";
+    private static final int VALID_USER_AGE = 20;
+    private static final int USER_EDGE_AGE = 18;
+    private static RegistrationServiceImpl registrationService;
+    private static StorageDaoImpl storageDao;
+    private User defaultValidUser;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        defaultValidUser = new User();
+        defaultValidUser.setAge(VALID_USER_AGE);
+        defaultValidUser.setLogin(VALID_LOGIN);
+        defaultValidUser.setPassword(VALID_PASSWORD);
+    }
+
+    @AfterEach
+    public void clearStorage() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void register_valid_ok() {
+        User actual = registrationService.register(defaultValidUser);
+        assertNotNull(actual);
+        assertEquals(defaultValidUser, actual);
+    }
+
+    @Test
+    void register_edgeAge_ok() {
+        defaultValidUser.setAge(USER_EDGE_AGE);
+        User actual = registrationService.register(defaultValidUser);
+        assertNotNull(actual);
+        assertEquals(defaultValidUser, actual);
+    }
+
+    @Test
+    void register_edgePassword_ok() {
+        defaultValidUser.setPassword(EDGE_PASSWORD);
+        User actual = registrationService.register(defaultValidUser);
+        assertNotNull(actual);
+        assertEquals(defaultValidUser, actual);
+    }
+
+    @Test
+    void register_edgeLogin_ok() {
+        defaultValidUser.setLogin(EDGE_LOGIN);
+        User actual = registrationService.register(defaultValidUser);
+        assertNotNull(actual);
```

Comment:
Actually probably this null assertion is not needed, just because after that we check if `expected` is equal to `actual`